master/internal/api_config_policies.go

```diff
@@ -35,29 +43,167 @@ func validatePoliciesAndWorkloadType(workloadType, configPolicies string) error
+			return status.Errorf(codes.InvalidArgument, globalPriorityErr)
+		}
+	}
+
+	// Validate against specific configurations
+	if expConfigPolicies != nil {
+		if err := a.validateExperimentConfig(
+			ctx, globalConfigPolicies, expConfigPolicies, constraints, workloadType,
+		); err != nil {
+			return status.Errorf(codes.InvalidArgument, fmt.Sprintf("invalid experiment config policy"+": %s.", err))
+		}
+	}
+
+	if ntscConfigPolicies != nil {
+		if err := a.validateNTSCConfig(ctx, globalConfigPolicies, ntscConfigPolicies, constraints, workloadType); err != nil {
+			return status.Errorf(codes.InvalidArgument, fmt.Sprintf("invalid ntsc config policy"+": %s.", err))
+		}
 	}
 
-	// TODO (request validation): Verify that configs do not violate constraints.
 	return nil
 }
 
-func parseConfigPolicies(configAndConstraints string) (tcps map[string]interface{},
-	invariantConfig *string, constraints *string, err error,
+func (a *apiServer) validateExperimentConfig(
+	ctx context.Context,
+	globalConfigPolicies *model.TaskConfigPolicies,
+	expConfigPolicies *expconf.ExperimentConfigV0,
+	constraints *model.Constraints, workloadType string,
+) error {
+	if expConfigPolicies.RawResources != nil {
+		if err := a.checkAgainstGlobalPriority(ctx, expConfigPolicies.RawResources.RawPriority, workloadType); err != nil {
+			return err
+		}
+		if err := checkConstraintConflicts(constraints, expConfigPolicies.RawResources.RawMaxSlots,
+			expConfigPolicies.RawResources.RawSlotsPerTrial, expConfigPolicies.RawResources.RawPriority); err != nil {
+			return err
+		}
+	}
+
+	return checkAgainstGlobalConfig(ctx, globalConfigPolicies, expConfigPolicies, nil, workloadType)
+}
+
+func (a *apiServer) validateNTSCConfig(
+	ctx context.Context,
+	globalConfigPolicies *model.TaskConfigPolicies,
+	ntscConfigPolicies *model.CommandConfig,
+	constraints *model.Constraints, workloadType string,
+) error {
+	if ntscConfigPolicies.Resources.Priority != nil {
+		if err := a.checkAgainstGlobalPriority(ctx, ntscConfigPolicies.Resources.Priority, workloadType); err != nil {
+			return err
+		}
+	}
+	if err := checkConstraintConflicts(constraints, ntscConfigPolicies.Resources.MaxSlots,
+		&ntscConfigPolicies.Resources.Slots, ntscConfigPolicies.Resources.Priority); err != nil {
+		return err
+	}
+
+	return checkAgainstGlobalConfig(ctx, globalConfigPolicies, nil, ntscConfigPolicies, workloadType)
+}
+
+func (a *apiServer) checkAgainstGlobalPriority(ctx context.Context, taskPriority *int, workloadType string) error {
+	if taskPriority == nil {
+		return nil
+	}
+
+	_, priorityEnabledErr := a.m.rm.SmallerValueIsHigherPriority()
+	if priorityEnabledErr != nil {
+		return fmt.Errorf("task priority is not supported in this cluster: %w", priorityEnabledErr)
+	}
+
+	_, globalPriorityExists, _ := configpolicy.GetPriorityLimit(ctx, nil, workloadType)
+	if globalPriorityExists {
+		return fmt.Errorf(globalPriorityErr)
```

Comment:
I don't understand the purpose of this check. We check elsewhere for conflicts between configs & constraints right? We also check elsewhere for overlap between workspace and global config policies. 

If this check isn't needed, the function name should probably be updated too.

Reply:
There might be a hole with checking for constraints set at both workspace and global level? I think this just checks for priority; we need to check slots too. Would it be possible to use something like `HaveAtLeastOneSharedDefinedField` for constraints too?